### PR TITLE
{vis} [goolf-1.4.10] ImageMagick-6.9.3-2 (REVIEW)

### PR DESCRIPTION
**Pull request by pescobar**
Imagemagick and deps. Deps extracted querying yum

```
[escobar@login18 lcms2-2.7]$ yum deplist ImageMagick-devel |grep dependency|grep -v .so|grep -v config|grep -v bin/sh|grep -vi imagemagic|sort -u
  dependency: bzip2-devel
  dependency: freetype-devel
  dependency: ghostscript-devel
  dependency: jasper-devel
  dependency: lcms-devel
  dependency: libjpeg-devel
  dependency: libtiff-devel
  dependency: libX11-devel
  dependency: libXext-devel
  dependency: libXt-devel
```

Alternative to https://github.com/hpcugent/easybuild-easyconfigs/pull/1522

### CODE REVIEW

**Comment by pescobar** [comment]
Test report by @pescobar
**SUCCESS**
Build succeeded for 21 out of 21 (3 easyconfigs in this PR)
Linux centos 6.5, Intel(R) Xeon(R) CPU           X5550  @ 2.67GHz, Python 2.6.6
See https://gist.github.com/2e685ed57fbb2bd64633 for a full test report.

**Comment by pescobar** [comment]
I reviewed the centos6 src rpm and these are the config options used in centos

```
%configure --enable-shared \
        --disable-static \
        --with-modules \
        --with-perl \
        --with-x \
        --with-threads \
        --with-magick_plus_plus \
        --with-gslib \
        --with-wmf \
        --with-webp \
        --with-openexr \
        --with-rsvg \
        --with-xml \
        --with-perl-options="INSTALLDIRS=vendor %{?perl_prefix} CC='%__cc -L$PWD/magick/.libs' LDDLFLAGS='-shared -L$PWD/magick/.libs'" \
        --without-dps \
        --without-gcc-arch \
        --with-jbig \
        --with-openjp2
```

not sure which ones I should add in the easyconfig....

**Comment by boegel** [comment]
@pescobar: I'm up for leaving it pretty much default, we can revisit later by adding configure options if something is missing?

**Comment by boegel** [comment]
@pescobar: custom `sanity_check_paths` in each of these would be great

File: easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+easyblock = 'ConfigureMake'
+
+name = 'ImageMagick'
+version = '6.9.3-2'
```
Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb]: this version seems to have disappeared, now there's only `6.9.3-3` available...

the version from #1522 (`6.9.0-9`) is also gone (now only `6.9.0-10` is there)

not sure how to handle this... is there an alternative download location?

Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb]: I found this, but not all versions are there: http://www.imagemagick.org/download/releases/

File: easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+homepage = 'http://www.imagemagick.org/'
+description = """ImageMagick is a software suite to create, edit, compose, or convert bitmap images"""
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://www.imagemagick.org/download/']
```
Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb]: @pescobar: please change to looks like this is a good alternative: `https://launchpad.net/imagemagick/main/%(version)s/+download/`, where all versions are available

Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-6.9.3-2-goolf-1.4.10.eb]: see https://launchpad.net/imagemagick/+download

**Comment by boegel** [comment]
@pescobar: remarks fixed in https://github.com/pescobar/easybuild-easyconfigs/pull/40, please review/merge?

**Comment by pescobar** [comment]
@boegel If you want to leave the default configure options then I would suggest to review the dependencies. IMHO it makes no sense to add ghostscript as a dependency and not using ```--with-gslib```

**Comment by boegel** [comment]
@pescobar: well, of course the configure flags that correspond with the dependencies should be there, I'd say that's obvious :-)

**Comment by ladc** [comment]
The sanity checks for libjpeg-turbo don't check out for me (that is, ``man`` -> ``share/man`` and ``libjpeg.%s`` -> ``lib/libjpeg.%s`` ). I also think there's a missing dep in LittleCMS: ``('libjpeg-turbo', '1.4.2'),``

File: easybuild/easyconfigs/l/libjpeg-turbo/libjpeg-turbo-1.4.2-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+dependencies = [
+    ('NASM', '2.11.08'),
+]
+
+configopts = "--with-jpeg8"
+runtest = "test"
+
+sanity_check_paths = {
+    'files': ['bin/cjpeg', 'bin/djpeg', 'bin/jpegtran', 'bin/rdjpgcom', 'bin/tjbench', 'bin/wrjpgcom',
+              'lib/libjpeg.a', 'libjpeg.%s' % SHLIB_EXT, 'lib/libturbojpeg.a', 'lib/libturbojpeg.%s' % SHLIB_EXT],
```
Comment (boegel) [inline easybuild/easyconfigs/l/libjpeg-turbo/libjpeg-turbo-1.4.2-goolf-1.4.10.eb]: missing `lib/` in 2nd entry (thanks @ladc)

File: easybuild/easyconfigs/l/libjpeg-turbo/libjpeg-turbo-1.4.2-goolf-1.4.10.eb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    'dirs': ['include', 'man', 'share'],
```
Comment (boegel) [inline easybuild/easyconfigs/l/libjpeg-turbo/libjpeg-turbo-1.4.2-goolf-1.4.10.eb]: this should have `share/man`, not both `man` (thanks @ladc)

File: easybuild/easyconfigs/l/LittleCMS/LittleCMS-2.7-goolf-1.4.10.eb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+homepage = 'http://www.littlecms.com/'
+description = """ Little CMS intends to be an OPEN SOURCE small-footprint color management engine, 
+ with special focus on accuracy and performance. """
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+source_urls = ['http://sourceforge.net/projects/lcms/files/lcms/%(version)s/']
+sources = ['lcms2-%(version)s.tar.gz']
+
```
Comment (boegel) [inline easybuild/easyconfigs/l/LittleCMS/LittleCMS-2.7-goolf-1.4.10.eb]: missing dep on `libjpeg-turbo` (thanks @ladc)

**Comment by boegel** [comment]
@pescobar: remarks fixed in https://github.com/pescobar/easybuild-easyconfigs/pull/43

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/19455bad2fcb22794f9c for a full test report.

**Comment by boegel** [comment]
Going in, thanks @pescobar, and @ladc for the feedback!

**Pull request merged**